k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -55,13 +57,11 @@ class EndpointsNamer(idPrefix: Path, mkApi: String => v1.NsApi) extends Namer {
     def get(name: String): NsCache = synchronized {
       activity.sample() match {
         case Activity.Ok(snap) =>
-          snap.get(name) match {
-            case Some(ns) => ns
-            case None =>
-              val ns = mkNs(name)
-              activity() = Activity.Ok(snap + (name -> ns))
-              ns
-          }
+          snap.getOrElse(name, {
+            val ns = mkNs(name)
+            activity() = Activity.Ok(snap + (name -> ns))
+            ns
+          })
```

Comment:
The signature for `Map[K,V].getOrElse` is effectively
```scala
def getOrElse(k: K, f: => V): V
```

In this case we happen to know that f is going to be called immediately, but generally speaking, this is not a good assumption that if i pass a closure to something that it will be called immediately (in the same thread, etc).  when I see a `synchronized` and then passing a closure to function, i need to stop and make sure that everything's safe ("ok, this isn't a Future, or a view, or something like that").  When there's no closure, it's easy for me to reason about without thinking about the specific types at play.